sdk/go/internal/engineconn/unix/unix.go

```diff
@@ -30,20 +30,10 @@ func (c *Unix) Addr() string {
 }
 
 func (c *Unix) Connect(ctx context.Context, cfg *engineconn.Config) (*http.Client, error) {
-	// FIXME: These are necessary for dagger-in-dagger but do not work.
-	// if cfg.Workdir != "" {
-	// 	return nil, errors.New("workdir not supported on unix hosts")
-	// }
-	// if cfg.ConfigPath != "" {
-	// 	return nil, errors.New("config path not supported on unix hosts")
-	// }
-	// if cfg.NoExtensions {
-	// 	return nil, errors.New("no extensions is not supported on unix hosts")
-	// }
 	return &http.Client{
 		Transport: &http.Transport{
 			DialContext: func(_ context.Context, _, _ string) (net.Conn, error) {
-				return net.Dial("unix", c.path)
+				return net.Dial("unix", c.u.Host+c.u.Path)
```

Comment:
Is this for extensions (e.g. this is talking to the session, not buildkit)?

In that case, shouldn't this be `http`?

Otherwise no way to connect to `dagger listen`, and this can't be done in Python (at all) / TypeScript (without hacks)

Reply:
Oh yeah so this is needed to get extensions to work, but you're right that it means we'd need to add unix to the nodejs SDK, so I was very confused as to why tests were passing after reading your comment...

Then realized we removed the yarn extension test a few days ago: https://github.com/dagger/dagger/commit/f74509711b55a10a4240c27a0f076cbd52965b32
* This was just so we didn't have to waste time maintaining extensions in the midst of the release work, which makes total sense to me

So yeah, Python and NodeJS extensions won't work currently with this. I can see if there's any other updates possible in the short term, but I think no matter what the long term fix will be the same as elsewhere: include engine-session functionality with each extension exec (probably via the shim), at which time this will disappear.

I'm inclined to just accept that nodejs+python extensions are broken at the moment and wait for that fix; given the test case was removed I think that is probably the state of them anyways. We at least still have Go extensions working to test the the project codepaths are basically in tact. Let me know if you disagree though.